RecoMuon/TrackerSeedGenerator/plugins/TSGForOIDNN.cc

```diff
@@ -0,0 +1,974 @@
+                               *(propagatorOpposite.get()),
+                               estimatorH,
+                               errorSFHitless,
+                               hitlessSeedsMadeMuS,
+                               numSeedsMade,
+                               out);
+
+        if (hitSeedsMade < maxHitSeeds__ && numSeedsMade < maxSeeds_)
+          makeSeedsFromHits(**it,
+                            tsosAtIP,
+                            *(propagatorAlong.get()),
+                            estimatorH,
+                            measurementTrackerH,
+                            hitSeedsMade,
+                            numSeedsMade,
+                            layerCount,
+                            out);
+
+        if (hitDoubletSeedsMade < maxHitDoubletSeeds__ && numSeedsMade < maxSeeds_)
+          makeSeedsFromHitDoublets(**it,
+                                   tsosAtIP,
+                                   *(propagatorAlong.get()),
+                                   estimatorH,
+                                   measurementTrackerH,
+                                   navSchool,
+                                   hitDoubletSeedsMade,
+                                   numSeedsMade,
+                                   layerCount,
+                                   out);
+      }
+      LogTrace("TSGForOIDNN") << "TSGForOIDNN:::produce: NumSeedsMade = " << numSeedsMade
+                              << " , layerCount = " << layerCount << std::endl;
+    }
+
+    for (std::vector<TrajectorySeed>::iterator it = out.begin(); it != out.end(); ++it) {
+      result->push_back(*it);
+    }
+
+  }  // L2Collection
+
+  edm::LogInfo(theCategory_) << "TSGForOIDNN::produce: number of seeds made: " << result->size();
+
+  iEvent.put(std::move(result));
+}
+
+//
+// Create seeds without hits on a given layer (TOB or TEC)
+//
+void TSGForOIDNN::makeSeedsWithoutHits(const GeometricSearchDet& layer,
+                                       const TrajectoryStateOnSurface& tsos,
+                                       const Propagator& propagatorAlong,
+                                       edm::ESHandle<Chi2MeasurementEstimatorBase>& estimator,
+                                       double errorSF,
+                                       unsigned int& hitlessSeedsMade,
+                                       unsigned int& numSeedsMade,
+                                       std::vector<TrajectorySeed>& out) const {
+  // create hitless seeds
+  LogTrace("TSGForOIDNN") << "TSGForOIDNN::makeSeedsWithoutHits: Start hitless" << std::endl;
+  std::vector<GeometricSearchDet::DetWithState> dets;
+  layer.compatibleDetsV(tsos, propagatorAlong, *estimator, dets);
+  if (!dets.empty()) {
+    auto const& detOnLayer = dets.front().first;
+    auto const& tsosOnLayer = dets.front().second;
+    LogTrace("TSGForOIDNN") << "TSGForOIDNN::makeSeedsWithoutHits: tsosOnLayer " << tsosOnLayer << std::endl;
```

Comment:
```suggestion
    LogTrace("TSGForOIDNN") << "TSGForOIDNN::makeSeedsWithoutHits: tsosOnLayer " << tsosOnLayer;
```
I think that `endl` is already a part of a LogTrace call

Reply:
Removed `endl` from all `LogTrace` calls.